lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -46,18 +57,17 @@ module.exports = function initialize(_agent, elastic, _moduleName, shim) {
  */
 function queryParser(params) {
   params = JSON.parse(params)
-
   const { collection, operation } = parsePath(params.path, params.method)
 
   // the substance of the query may be in querystring or in body.
   let queryParam = {}
-  if (typeof params.querystring === 'object' && Object.keys(params.querystring).length > 0) {
+  if (isSimpleObject(params.querystring) && Object.keys(params.querystring).length > 0) {
     queryParam = params.querystring
   }
   // let body or bulkBody override querystring, as some requests have both
-  if (typeof params.body === 'object' && Object.keys(params.body).length > 0) {
+  if (isSimpleObject(params.body) && Object.keys(params.body).length > 0) {
     queryParam = params.body
-  } else if (typeof params.bulkBody === 'object' && Object.keys(params.bulkBody).length > 0) {
+  } else if (isSimpleObject(params.bulkBody) && Object.keys(params.bulkBody).length > 0) {
```

Comment:
With this repetition I would add `isNonEmptyObject` util:

`const isNonEmptyObject = (thing) => isSimpleObject(thing) && Object.keys(thing).length > 0`